### PR TITLE
Aweful RzAsm <-> RzAnalysis pointer sharing and Hexagon unify

### PR DESCRIPTION
**Pull request by Rot127**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

**Hacks**

Attempt to share pointers between `RzAsm` and `RzAnalysis` without any `RzArch`.

The assumption made here is that `RzAsm` is always initialized. Also without any `RzAnalysis`. But `RzAnalysis` is never initialized without a companion `RzAsm`.
Hence a plugin requiring a common state, always accesses `RzAsm->plugin_data`, no matter if it was called from `RzAnalysis` or `RzAsm`.

`RzAnalysis` holds the file buffer. So if this one is present, the plugin should initialize a buffer around it. If it is not there, it should make a byte buffer around the bytes passed to the function.

This mimics future behavior of `RzArch` which will get either an `IO` or `Bytes` buffer.

**Plugin configs**

With this change there is also the need to refactor the plugin specific configs.
I designed them badly before. And they couldn't be used by a plugin without a static variable.

Plugin specific configs are hold now in a hash mpa in RzCore. The reason to have them separately and not in the `RzCore->config` is because plugins can be the _owner_ of them. Meaning the `config_set` functions will pass a pointer with the plugin state. Not the core state. This is required if plugins have specific configuration options they need to know, but have no access to `RzCore` to get them.

It adds luckily only a few more lines to handle them in `cmd_eval.c` and for the auto-complete.

**Hexagon changes**

- Moves the state from a static variable into `RzAsm->plugin_data`.
- Always disassembles the context around an instruction to know the packet properties the instruction is in.

**Test plan**

For plugin configs
Otherwise: all green with fuzz-dist.

**Closing issues**

None. Probably opens a few more.

### CODE REVIEW

**Review by XVilka — COMMENTED** [review]
The changes are good. I am only in doubt about those hacks. Let's see what others say.

File: librz/core/core.c [hunk 1]
```diff
@@ -1,6 +1,8 @@
 // SPDX-FileCopyrightText: 2009-2020 pancake <pancake@nopcode.org>
 // SPDX-License-Identifier: LGPL-3.0-only
 
+#include "rz_config.h"
+#include "rz_util/ht_sp.h"
```
Comment (wargio) [inline librz/core/core.c]:
```suggestion
#include <rz_config.h>
#include <rz_util/ht_sp.h>
```

**Review by wargio — APPROVED** [review]
LGTM. Hopefully this hack will go away with RzArch

**Review by XVilka — APPROVED** [review]
LGTM but only as a temporary measure. We should not make a new release with this hack.

**Review by wargio — APPROVED** [review]

**Pull request merged**